src/builder/mod.rs

```diff
@@ -12,9 +12,16 @@ use crate::http::CacheHttp;
 #[cfg(feature = "http")]
 use crate::internal::prelude::*;
 
+/// Common trait for all HTTP request builders in this module.
+///
+/// Will have a [`serde::Serialize`]` + `[`serde::Deserialize`] bound in the future, because
+/// builders need to be serializable in order to send them to the Discord API and
+/// [`serde::Deserialize`] is useful to be able to store builders locally.
 #[cfg(feature = "http")]
 #[async_trait::async_trait]
-pub trait Builder {
+pub trait Builder<'de>: serde::Serialize + serde::Deserialize<'de> {
```

Comment:
```suggestion
pub trait Builder: serde::Serialize + serde::DeserializeOwned {
```
To avoid adding extra complexity, I think bounding on `DeserializeOwned` is better. None of the serializable fields in builders have any lifetimes, anyways.